### PR TITLE
chore: bump all versions to 1.8.0 and remove old migrations.

### PR DESCRIPTION
**Pull request by dandanlen**
I branched release/1.7 so am bumping all versions here.

### CODE REVIEW

File: engine-upgrade-utils/src/lib.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ pub mod build_helpers;
 // relevant crates.
 // Should also check that the compatibility function below `args_compatible_with_old` is correct.
 pub const OLD_VERSION: &str = "1.6.7";
```
Comment (kylezs) [inline engine-upgrade-utils/src/lib.rs:13]: old version should be 1.7 now

File: engine/multisig/Cargo.toml [hunk 2]
```diff
@@ -43,7 +43,7 @@ public-ip = "0.2.2"
 curve25519-dalek = { version = "4.1.3", features = ["serde"] }
 typenum = "1.15"
 schnorrkel = "0.11.4"
-rayon = "1.7.0"
```
Comment (kylezs) [inline engine/multisig/Cargo.toml:46]: 👀

**Review by kylezs — APPROVED** [review]

File: engine/src/state_chain_observer/client/extrinsic_api/signed/submission_watcher.rs [hunk 3]
```diff
@@ -647,3 +657,8 @@ impl<'a, 'env, BaseRpcClient: base_rpc_api::BaseRpcApi + Send + Sync + 'static>
 		}
 	}
 }
+
+#[test]
```
Comment (kylezs) [inline engine/src/state_chain_observer/client/extrinsic_api/signed/submission_watcher.rs:661]: accidentally left this in?

File: api/bin/chainflip-broker-api/src/main.rs [hunk 4]
```diff
@@ -10,19 +10,27 @@ use chainflip_api::{
...
 	server::ServerBuilder,
+	types::ErrorCode,
 };
 use std::{
 	path::PathBuf,
 	sync::{atomic::AtomicBool, Arc},
 };
 use tracing::log;
 
+fn to_error_object(error: impl core::fmt::Display) -> jsonrpsee::types::error::ErrorObjectOwned {
```
Comment (dandanlen) [inline api/bin/chainflip-broker-api/src/main.rs:26]: TODO: make explicit anyhow error.

File: api/bin/chainflip-lp-api/src/main.rs [hunk 5]
```diff
@@ -229,11 +230,33 @@ impl RpcServerImpl {
...
+			ErrorObject::owned(
+				ErrorCode::InternalError.code(),
+				"Internal rpc client error",
+				None::<()>,
+			)
+		},
+	}
+}
+
+fn to_error_object(error: impl core::fmt::Display) -> jsonrpsee::types::error::ErrorObjectOwned {
```
Comment (dandanlen) [inline api/bin/chainflip-lp-api/src/main.rs:252]: TODO: Same as above

**Review by kylezs — APPROVED** [review]

File: state-chain/runtime/src/lib.rs [hunk 6]
```diff
@@ -1272,14 +1268,7 @@ type PalletMigrations = (
...
-	VersionedMigration<
-		pallet_cf_broadcast::Pallet<Runtime, SolanaInstance>,
-		SerializeSolanaBroadcastMigration,
-		8,
-		9,
-	>,
+type MigrationsForV1_8 = (
 	// For clearing all Solana Egress Success election votes, and migrating Solana ApiCall to the
 	// newer version.
 	VersionedMigration<
```
[the remark this answers is not in the document]
Comment (dandanlen) [reply state-chain/runtime/src/lib.rs:1274]: Because it was added after 1.7 was branched.

**Comment by dandanlen** [comment]
FYI @kylezs 
I've disabled auto-merge because (i assume) it won't pass the merge queue anyway until #5346 passes. And then there are other PRs that need to be merged into 1.7 (with migrations) so I think this PR will need further adjustment before it's ready. Will dismiss your review so we don't accidentally merge prematurely.

**Review by kylezs — APPROVED** [review]

**Pull request merged**